commit f7510549b109dcf950406b283956df84f56147fa
Author: Denis Barbier <bouzim@gmail.com>
Date:   Wed Apr 6 22:30:55 2011 +0200

    Bug fix: Text doesn't get displayed in 6.5
    
    Submitted upstream: http://www.opencascade.org/org/forum/thread_20101/
    
    Based on a patch provided on the opencascade forum by Venugopal Gudimetla.

In OCCT 6.5.0, code was:
  Display * disp = XOpenDisplay("localhost:0.0");
  if ( !disp ) 
  {
    cout << "Display is NULL!" << endl;
    return ;
  }

It has been changed in 0CCT 6.5.1 to:
  Display * disp = XOpenDisplay("localhost:0.0");
  if (!disp) 
  {
    disp = XOpenDisplay(":0.0");
    if (!disp)
    {
      cout << "Display is NULL!" << endl;
      return ;
    }
  }
and is described in issue 22105 of their release notes as a way to fix
remote displays.

But this fix is wrong, it won't work for SSH connections, for instance.
It is much more robust to rely on DISPLAY environment variable, so we keep
this patch.

Index: oce/src/OSD/OSD_FontMgr.cxx
===================================================================
--- oce.orig/src/OSD/OSD_FontMgr.cxx
+++ oce/src/OSD/OSD_FontMgr.cxx
@@ -236,7 +236,7 @@
 #ifndef WNT
   StringList dirs;
   Handle(TCollection_HAsciiString) str = new TCollection_HAsciiString;
-  Display * disp = XOpenDisplay("localhost:0.0");
+  Display * disp = XOpenDisplay(NULL);
 
   if (!disp) 
   {
